File: TeamProject.Models/Models/PropertyImage.cs
using System.ComponentModel.DataAnnotations;

namespace TeamProject.Models.Models;

public class PropertyImage
{
    [Key]
    public int Id { get; set; }

    public string ImageUrl { get; set; } = string.Empty;

    public int PropertyId { get; set; }
    public Property Property { get; set; } = null!;
}

File: src/lib/server/db/entities/property-image.ts
import {
  Column,
  Entity,
  JoinColumn,
  ManyToOne,
  PrimaryGeneratedColumn,
  type Relation,
} from "typeorm";
import { Property } from "./property";

@Entity({ name: "propertyImages" })
export class PropertyImage {
  @PrimaryGeneratedColumn("uuid")
  id!: string;

  @Column({ type: "text" })
  imageUrl!: string;

  @Column({ type: "text" })
  propertyId!: string;

  // many images may belong to one property
  @ManyToOne(
    () => Property,
    (property) => property.images,
    { onDelete: "CASCADE" },
  )
  @JoinColumn({ name: "propertyId" })
  property!: Relation<Property>;
}
